linodecli/baked/parsing.py

```diff
@@ -0,0 +1,184 @@
+"""
+This module contains logic related to string parsing and replacement.
+"""
+
+import functools
+import re
+from html import unescape
+from typing import List, Tuple
+
+# Sentence delimiter, split on a period followed by any type of
+# whitespace (space, new line, tab, etc.)
+REGEX_SENTENCE_DELIMITER = re.compile(r"\.(?:\s|$)")
+
+# Matches on pattern __prefix__ at the beginning of a description
+# or after a comma
+REGEX_TECHDOCS_PREFIX = re.compile(r"(?:, |\A)__([\w-]+)__")
+
+# Matches on pattern [link title](https://.../)
+REGEX_MARKDOWN_LINK = re.compile(r"\[(?P<text>.*?)]\((?P<link>.*?)\)")
+
+MARKDOWN_RICH_TRANSLATION = [
```

Comment:
I was originally trying to use the Rich `Markdown` renderable but there were a bunch of small issues relating to grouping and styles that heavily complicated things. For the time being this should work 🙂